library/ssl_tls.c

```diff
@@ -7309,10 +7309,13 @@ static int ssl_parse_certificate_verify( mbedtls_ssl_context *ssl,
     mbedtls_x509_crt *ca_chain;
     mbedtls_x509_crl *ca_crl;
 
-    ssl->handshake->peer_authenticated = MBEDTLS_SSL_FI_FLAG_UNSET;
     if( authmode == MBEDTLS_SSL_VERIFY_NONE )
```

Comment:
By the way, I think we want a larger Hamming distance between `VERIFY_NONE`, `VERIFY_OPTIONAL` and `VERIFY_REQUIRED` too.